server/src/main/java/org/opensearch/action/support/AutoCreateIndex.java

```diff
@@ -109,9 +105,6 @@ public boolean shouldAutoCreate(String index, ClusterState state) {
         if (autoCreate.autoCreateIndex == false) {
             throw new IndexNotFoundException("[" + AUTO_CREATE_INDEX_SETTING.getKey() + "] is [false]", index);
         }
-        if (dynamicMappingDisabled) {
```

Comment:
I think we should warn here, similarly to `MapperService`:

```
if (INDEX_MAPPER_DYNAMIC_SETTING.exists(settings)) {
            deprecationLogger.deprecate(
                index().getName() + INDEX_MAPPER_DYNAMIC_SETTING.getKey(),
                "Index [{}] has setting [{}] that is not supported in OpenSearch, its value will be ignored.",
                index().getName(),
                INDEX_MAPPER_DYNAMIC_SETTING.getKey()
            );
        }
```

Reply:
This is dead code. The `settings` passed into the constructor are the node settings.

If you try to add `index.mapper.dynamic: false` into `opensearch.yml` the cluster fails to startup with:

```
java.lang.IllegalArgumentException: node settings must not contain any index level settings
	at org.opensearch.common.settings.SettingsModule.<init>(SettingsModule.java:172)
	at org.opensearch.node.Node.<init>(Node.java:593)
	at org.opensearch.node.Node.<init>(Node.java:419
        ...
```

Reply:
Thanks @cwperks !